addons/@mlabs-haskell/godot-cardano/src/cardano.gd

```diff
@@ -11,42 +11,48 @@ signal got_wallet
 
 var provider: Provider
 var wallet: Wallet
-var tx_builder: TxBuilder
+var _protocol_params: ProtocolParameters
+var _era_summaries: Array[Provider.EraSummary]
 
 func _init(provider_: Provider) -> void:
 	self.provider = provider_
 	self.wallet = null
-	self.tx_builder = null
 	add_child(provider)
 	if provider.got_protocol_parameters.connect(_on_got_protocol_parameters) == ERR_INVALID_PARAMETER:
 		push_error("Failed to connect provider's 'got_protocol_parameters' signal ")
+	if provider.got_era_summaries.connect(_on_got_era_summaries) == ERR_INVALID_PARAMETER:
+		push_error("Failed to connect provider's 'got_era_summaries' signal ")
 
 func _ready() -> void:
 	@warning_ignore("redundant_await")
-	var _params := await provider.get_protocol_parameters()
+	var _params := await provider._get_protocol_parameters()
```

Comment:
Is it OK to await during `_ready`? I think this will introduce latency in the scene tree during initialization. It also seems redundant since the parameters and era summaries are being saved asynchronously below in the callbacks `_on_got_protocol_parameters` and `_on_got_era_summaries`.

I would remove the `await`s and let the callbacks below handle the tasks' completion.

Reply:
On the other hand, I think all of these overridable functions (`_ready`, `_process`, etc) are synchronous, so I'm not sure they even stop their execution when an `await` instruction is found.